modules/reindex/src/main/java/org/elasticsearch/index/reindex/RestReindexAction.java

```diff
@@ -151,11 +152,12 @@ static RemoteInfo buildRemoteInfo(Map<String, Object> source) throws IOException
         String scheme = hostMatcher.group("scheme");
         String host = hostMatcher.group("host");
         int port = Integer.parseInt(hostMatcher.group("port"));
+        Map<String, String> headers = extractStringStringMap(remote, "headers");
         if (false == remote.isEmpty()) {
             throw new IllegalArgumentException(
                     "Unsupported fields in [remote]: [" + Strings.collectionToCommaDelimitedString(remote.keySet()) + "]");
         }
-        return new RemoteInfo(scheme, host, port, queryForRemote(source), username, password);
+        return new RemoteInfo(scheme, host, port, queryForRemote(source), username, password, headers);
```

Comment:
Funky proxies is what I was thinking. I wrote it to work with that funny fake security plugin that I ended up removing but I kept the headers support because I thought that it might be useful to work with proxies between the two Elasticsearch clusters.